crates/biome_js_analyze/src/lint/nursery/no_static_element_id.rs

```diff
@@ -0,0 +1,121 @@
+    /// React.createElement("div", { id: "foo" });
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// const id = useId();
+    /// <div id={id}>bar</div>;
+    /// ```
+    ///
+    /// ```jsx
+    /// const id = useId();
+    /// React.createElement("div", { id });
+    /// ```
+    ///
+    pub NoStaticElementId {
+        version: "next",
+        name: "noStaticElementId",
+        language: "jsx",
+        recommended: true,
+        severity: Severity::Error,
+        domains: &[RuleDomain::React],
+    }
+}
+
+declare_node_union! {
+    pub IdProp = JsxAttribute | JsPropertyObjectMember
+}
+
+declare_node_union! {
+    pub NoStaticElementIdQuery = AnyJsxElement | JsCallExpression
+}
+
+impl NoStaticElementIdQuery {
+    fn find_id_attribute(&self, model: &SemanticModel) -> Option<IdProp> {
+        match self {
+            Self::AnyJsxElement(jsx) => jsx.find_attribute_by_name("id").map(IdProp::from),
+            Self::JsCallExpression(expression) => {
+                let react_create_element =
+                    ReactCreateElementCall::from_call_expression(expression, model)?;
+                react_create_element
+                    .find_prop_by_name("id")
+                    .map(IdProp::from)
+            }
+        }
+    }
+}
+
+impl Rule for NoStaticElementId {
+    type Query = Semantic<NoStaticElementIdQuery>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let model = ctx.model();
+        let id_attribute = node.find_id_attribute(model)?;
+
+        match id_attribute {
+            IdProp::JsxAttribute(jsx_attribute) => {
+                let jsx_any_attribute_value = jsx_attribute.initializer()?.value().ok()?;
+                if matches!(jsx_any_attribute_value, AnyJsxAttributeValue::JsxString(_)) {
+                    return Some(());
+                }
+                None
+            }
+            IdProp::JsPropertyObjectMember(js_object_member) => {
+                let expression = js_object_member.value().ok()?;
+                if matches!(expression, AnyJsExpression::AnyJsLiteralExpression(_)) {
+                    return Some(());
+                }
+                None
+            }
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "`id` attribute should not be static string literal. Generate unique IDs using `useId`."
+                },
+            )
+            .note(markup! {
+                "In React, if you hardcode IDs and use the component multiple times, it can lead to duplicate IDs in the DOM. So it's better to generate unique IDs using `useId`."
```

Comment:
you can use `<Emphasis>` tag instead of back quotation ` here. please check some other rules